packages/mcp/src/secrets/provider-configs.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Configuration types for secret providers in the MCP Funnel system.
+ *
+ * This module defines configuration structures for various secret providers
+ * that can be used to load environment variables and secrets from different sources.
+ * The discriminated union design allows for type-safe configuration and
+ * easy extension with additional provider types in the future.
+ */
+
+/**
+ * Configuration for the dotenv secret provider.
+ * Loads secrets from a .env file on the filesystem.
+ */
+interface ProviderConfigBase {
+  /**
+   * Optional provider identifier used for registration and diagnostics.
+   */
+  name?: string;
+}
+
+export interface DotEnvProviderConfig extends ProviderConfigBase {
+  /** Provider type discriminator */
+  type: 'dotenv';
+  /** Provider-specific configuration */
+  config: {
+    /**
+     * Path to the .env file to load.
+     * Can be relative (to the config file) or absolute.
+     * @example '.env', '/absolute/path/.env', './configs/.env.local'
+     */
+    path: string;
+    /**
+     * File encoding to use when reading the .env file.
+     * @default 'utf-8'
+     */
+    encoding?: string;
+  };
+}
+
+/**
+ * Configuration for the process environment secret provider.
+ * Loads secrets from the current process environment variables.
+ */
+export interface ProcessEnvProviderConfig extends ProviderConfigBase {
+  /** Provider type discriminator */
+  type: 'process';
+  /** Provider-specific configuration */
+  config: {
+    /**
+     * Only include environment variables that start with this prefix.
+     * The prefix will be stripped from the variable name when exposed.
+     * @example 'MCP_' - includes MCP_API_KEY as API_KEY
+     */
+    prefix?: string;
+    /**
+     * Allowlist of specific environment variable names to include.
+     * Takes precedence over prefix filtering when both are specified.
+     * When used alongside {@link prefix}, the entries must contain the full
+     * environment key (for example `MCP_API_KEY`) because matching happens
+     * before the prefix is stripped.
+     * @example ['API_KEY', 'DATABASE_URL']
```

Comment:
The note explains that allowlist entries must include the full environment key when a prefix is configured, but the `@example` still shows stripped names. Copying this example under `prefix: 'MCP_'` drops every variable because the filter runs before the prefix is removed (see `process-env-provider.ts` lines 76–83). Please update the example to use prefixed names like `['MCP_API_KEY', 'MCP_DATABASE_URL']` so the docs match the implementation.

Reply:
/fix